src/trace.c

```diff
@@ -1040,6 +1046,37 @@ trace_add_path (const void *parent,
 	if (! path_hash)
 		path_hash = NIH_MUST (nih_hash_string_new (NULL, 2500));
 
+	/* Start checking the file type to see if it can be
+	 * resolved down to some ordinary file.
+	 */
+	if (lstat (pathname, &statbuf) < 0) {
+		return 0;
+	}
+
+	if (! S_ISREG (statbuf.st_mode)) {
+		/* Not an ordinary file. check if it is a symlink,
+		 * and resolve to a file that symlink points to.
+		 */
+		if (S_ISLNK (statbuf.st_mode)) {
+			if (! realpath (pathname, resolved_pathname)) {
+				return 0;
+			}
+
+			pathname = resolved_pathname;
```

Comment:
We're not updating `path_hash` along with `pathname`, so it loos like we're inserting a wrong nih_hash entry.

Reply:
I was under the assumption that it is better to group up to a real path in case the file was symlink. I have updated the code, thank you!

Reply:
oh, sorry. I just misunderstood the semantics of  `nih_hash_add`. I thought the first `path_hash` was a pointer to a hashed string instead of a hashmap. Never mind, thanks.